catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -1488,24 +1488,23 @@ protected ResourceResponse getResource(ResourceRequest resourceRequest, boolean
                 resourceSourceName = resolvedSourceId;
             }
 
-            String key;
-            try {
-                key = new CacheKey(metacard, resourceRequest).generateKey();
-            } catch (Exception e1) {
-                LOGGER.error("resource not found", e1);
-                throw new ResourceNotFoundException(e1);
+            String key = new CacheKey(metacard, resourceRequest).generateKey();
+            if (key == null) {
+                LOGGER.error("Resource not found", resourceRequest);
+                throw new ResourceNotFoundException("Resource not found : " + resourceRequest);
             }
+
             if (productCache != null && productCache.containsValid(key, metacard)) {
-                try {
-                    Resource resource = productCache.getValid(key, metacard);
+                Resource resource = productCache.getValid(key, metacard);
```

Comment:
In the MetacardResourceSizePlugin we changed the TCCL, do we need to do the same thing here? If not, why are we doing it in the plugin?